crates/lib3h/src/gateway/gateway_dht.rs

```diff
@@ -96,7 +97,33 @@ impl P2pGateway {
                 unreachable!();
             }
             DhtRequestToParent::RequestEntry(_) => {
-                unreachable!();
+                self.endpoint_self.request(
+                    span,
+                    GatewayRequestToParent::Dht(payload),
+                    Box::new(|me, response| {
+                        trace!("Received requestEntry response in Gateway");
+                        let dht_response = match response {
+                            GhostCallbackData::Response(Ok(
+                                GatewayRequestToParentResponse::Dht(d),
+                            )) => d,
+                            GhostCallbackData::Response(Err(e)) => {
+                                panic!("Got error on GatewayRequest DHT: {:?} ", e);
+                            }
+                            GhostCallbackData::Timeout(bt) => {
+                                panic!("Got timeout on GatewayRequest DHT: {:?}", bt);
+                            }
+                            _ => panic!("Got wrong response type"),
```

Comment:
I've been squishing all these failure cases into one arm for brevity / readability:

```rust
                        let dht_response = match response {
                            GhostCallbackData::Response(Ok(
                                GatewayRequestToParentResponse::Dht(d),
                            )) => d,
                            _ => panic!("invalid response type: {:?}", response),
                        }
```

I like how you are allowed to access the original variable in the catch-all arm if it fails to destructure : )